### PR TITLE
Run tar/gzip at compile priority

### PR DESCRIPTION
**Pull request by derekgorcz**
The priority(nice value) of the local icecc will be used for tar when extracting the environment on a remote server instead of the priority of the iceccd process that is running remotely.

### CODE REVIEW

File: client/remote.cpp [hunk 1]
```diff
@@ -410,7 +412,13 @@ static int build_remote_int(CompileJob &job, UseCSMsg *usecs, MsgChannel *local_
...
             }
 
-            EnvTransferMsg msg(job.targetPlatform(), job.environmentVersion());
+            int extract_priority = getpriority(PRIO_PROCESS, 0);
+            if (-1 == extract_priority){
+                log_perror("Unable to determine process priority");
+                extract_priority = 0;
+            }
+
+            EnvTransferMsg msg(job.targetPlatform(), job.environmentVersion(), extract_priority);
```
Comment (derekgorcz) [inline client/remote.cpp]: The EnvTransferMsg will now include the 'extract_priority'.

File: tests/test.sh [hunk 2]
```diff
@@ -854,6 +854,74 @@ zero_local_jobs_test()
...
+{
+    echo Running environment_extraction_priority_test test.
+    reset_logs remote "$@"
+
+    mkdir "$testdir/new_test/" &> /dev/null
+
+    test_priorities=(10 12 15)
+    for compile_priority in ${test_priorities[@]}; do
+        kill_daemon remoteice1
+        start_iceccd remoteice1 -p 10246 -m 2
```
Comment (derekgorcz) [inline tests/test.sh]: We have to restart the daemon here to clear it's cache which contains the extracted environment from a previous build. If this is not done we won't see tar executing.

File: tests/test.sh [hunk 3]
```diff
@@ -854,6 +854,74 @@ zero_local_jobs_test()
...
+            echo Icecream not ready, aborting.
+            stop_ice 0
+            abort_tests
+        fi
+        sleep 5 # Give the scheduler time to get everything set up
+        flush_logs
+        ICECC_TEST_SOCKET="$testdir"/socket-localice ICECC_TEST_REMOTEBUILD=1 ICECC_PREFERRED_HOST=remoteice1 ICECC_DEBUG=debug ICECC_LOGFILE="$testdir"/icecc.log \
+            nice -n ${compile_priority} $valgrind "${icecc}" $GXX -Wall -Werror -g -c plain.cpp -o "$testdir/new_test/plain.o" 2>"$testdir"/stderr.remoteice &
+        compile_pid=$!
+        for wait_count in $(seq 1 100); do
```
Comment (derekgorcz) [inline tests/test.sh]: This loop polls for the existence of the tar process.

File: client/remote.cpp (hunk 1 above)
Comment (HenryMiller1) [inline client/remote.cpp]: Also, compile priority is already controlled by the daemon, I think that is the value you use be using.

File: daemon/environment.cpp [hunk 4]
```diff
@@ -589,6 +591,10 @@ pid_t start_install_environment(const std::string &basename, const std::string &
         log_perror("dup2 failed");
     }
 
+    if (-1 == setpriority(PRIO_PROCESS, 0, extract_priority)){
```
Comment (HenryMiller1) [inline daemon/environment.cpp]: serve.cpp handle_connection() we use 
'''
int niceval = nice(nice_level);
    if (niceval == -1) {
        log_warning() << "failed to set nice value: " << strerror(errno)
                      << endl;
    }
'''

Seems like we need to be consistent

File: tests/test.sh (hunk 3 above)
Comment (HenryMiller1) [inline tests/test.sh]: This looks like a race condition.  On a loaded machine tar might not start in the expected time, while on a fast machine tar might finish before the test gets here.

I don't know how to test this code... I don't think it is possible to do it with the current style of tests we have.  I think a unit test of some sort is needed. (we currently are not structured to unit test the code in that way)

**Comment by derekgorcz** [comment]
Using a slightly modified version of the removed test, I have been able to verify that the 'tar' process is executed using the 'nice' value which was provided to the remoteice1 daemon using the '--nice' option.
https://travis-ci.org/derekgorcz/icecream/jobs/234653412#L1070
https://github.com/derekgorcz/icecream/commits/test-build

**Pull request merged**